Seagate/cortx-manager
# EOS-20366: BE: Expose Admin Role

# Backend

# Problem Statement

[EOS-20366](https://jts.seagate.com/browse/EOS-20366): Expose Admin Role as along with Manage and monitor. Remove limitation of creation of only one Admin user, add sorting and fltering capability based on role and user name.

## Design

- expose POST /api/v2/csm/users endpoint as public,
- amend handling public request to search for the session if the token is passed and valid,
- implement admin specific checks for creation/deletion in Users service
- remove admin user specific endpoints
- update User model to support only one role instead of list of roles
- merge manage role permissions into the admin role


## Coding

- Coding conventions are followed and code is consistent. Yes
- Confirm All CODACY errors are resolved. Yes

## Testing

- Confirm that Test Cases are added (for both the cases, fix and feature). Yes
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes
- Confirm Testing was performed with installed RPM. Yes
- Confirm all the ut/regression/smoke test has been performed Yes

## Checklist

- _PR is self-reviewed? Yes/No?_
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_
        - _Deployment test : Done/Not?_
        - _UT/smoke test: Done/Not?_
        - _Jira number added to PR: Yes/No?_
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_
